lib/gather/collector.rb

```diff
@@ -40,18 +41,33 @@ def self.physical_data
       data[:model] = `dmidecode -s system-product-name`.chomp
       data[:bios] = `dmidecode -s bios-version`.chomp
       data[:serial] = `dmidecode -s system-serial-number`.chomp
-      data[:ram] = `grep MemTotal /proc/meminfo | awk '{print $2}'`.chomp # RAM measured in kB
+
+      # Get RAM info
+
+      data[:ram] = {}
+      ram_info = `dmidecode -q -t 17`.split('Memory Device')[1..]
+      data[:ram][:total_capacity] = si_to_bytes(between(`dmidecode -t 16`, 'Maximum Capacity: ', "\n")) / (1000.0 ** 3)
+      data[:ram][:units] = ram_info&.size
+      data[:ram][:capacity_per_unit] = data[:ram][:total_capacity] / data[:ram][:units]
+      data[:ram][:ram_data] = {}
+      ram_info&.each&.with_index() do |device, index|
+        data[:ram][:ram_data]["RAM#{index}"] = { form_factor: between(device, 'Form Factor: ', "\n"),
+                                                 size: si_to_bytes(between(device, 'Size: ', "\n")) / (1000.0 ** 3),
+                                                 locator: between(device, "\tLocator: ", "\n") }
+      end
 
       # Get processor info
 
       data[:cpus] = {}
       proc_info = `dmidecode -q -t processor`.split('Processor Information')[1..]
+      data[:cpus][:units] = proc_info&.size
+      data[:cpus][:cores_per_cpu] = [between(proc_info&.first, 'Thread count: ', "\n").to_i, 1].max
```

Comment:
This is incorrect, `Thread Count` includes cores from hyperthreading, we should be using `Core Count`

Reply:
Fixed 981f3ff